File: src/main/java/fr/vlegall/sochief/service/difficulty/RecipeDifficultyMapper.java
package fr.vlegall.sochief.service.difficulty;

import fr.vlegall.sochief.contracts.common.NamedIdDto;
import fr.vlegall.sochief.model.recipe.RecipeDifficulty;
import org.springframework.stereotype.Component;

@Component
public class RecipeDifficultyMapper {
    public NamedIdDto toDto(RecipeDifficulty recipeDifficulty) {
        return new NamedIdDto(recipeDifficulty.getId(), recipeDifficulty.getName());
    }
}

File: src/main/kotlin/fr/vlegall/sochief/service/difficulty/RecipeDifficultyMapper.kt
package fr.vlegall.sochief.service.difficulty

import fr.vlegall.sochief.model.recipe.RecipeDifficulty
import fr.vlegall.sochief.contracts.common.NamedIdDto
import org.springframework.stereotype.Component

@Component
class RecipeDifficultyMapper {
    fun toDto(recipeDifficulty: RecipeDifficulty): NamedIdDto =
        NamedIdDto(requireNotNull(recipeDifficulty.id), requireNotNull(recipeDifficulty.name))
}